### PR TITLE
docs(api): Sort path+body+body params by required

### PR DESCRIPTION
**Pull request by schew2381**
## Summary
Before we allowed parameters to come in any order, but we now order them by placing required params first. We can easily due this with `path`+`query` params using drf-spec, but `body` params are stored as a dictionary with implicit ordering due to python dicts now having ordering in 3.6+. 

To get around this, we parse the `body` params in the postprocess hook and explicitly order by converting it to an OrderedDict, placing required params at the front.

Note that other than required parameters, the ordering remains the same as specified in `@extend_schema` decorator. For `path`+`query` params, this the order of the `parameters` list kwarg. For `body` params, this is the order of the serializer fields that are specified in the serializer that is passed to the `request` kwarg

## Body Params
Nothing is modified by the `body` params change

## Path+Query Params
The endpoints that are modified by sorting `path`+`query` params are:
1. [Retrieve Event Counts for an Organization (v2)](https://docs.sentry.io/api/organizations/retrieve-event-counts-for-an-organization-v2/) - `groupBy` & `field` are required
### Before
Screenshot 2023-09-21 at 12 46 34 AM

### After
Screenshot 2023-09-21 at 12 53 23 AM

3. [Query Discover Events in Table Format](https://docs.sentry.io/api/discover/query-discover-events-in-table-format/) - `field` is required
### Before
Screenshot 2023-09-21 at 12 47 01 AM

### After
Screenshot 2023-09-21 at 12 52 34 AM

### CODE REVIEW

File: src/sentry/apidocs/api_ownership_stats_dont_modify.json
Comment (schew2381) [inline src/sentry/apidocs/api_ownership_stats_dont_modify.json:1]: You can ignore this file change, it's here b/c the github action to generate this JSON file is not being run in other PRs, so the changes show up here

File: src/sentry/apidocs/hooks.py [hunk 1]
```diff
@@ -197,18 +198,44 @@ def custom_postprocessing_hook(result: Any, generator: Any, **kwargs: Any) -> An
...
+                    if "multipart/form-data" in content:
+                        schema = content["multipart/form-data"]["schema"]
+                    else:
+                        schema = content["application/json"]["schema"]
+
+                    # Required params are stored in a list and not in the param itself
+                    required = set(schema.get("required", []))
+                    if required:
+                        # Explicitly sort body params by converting the dict to an ordered dict
+                        schema["properties"] = OrderedDict(
```
Comment (schew2381) [inline src/sentry/apidocs/hooks.py:229]: I think explicitly using an `OrderedDict` is better than sorting the original dict and relying on our python version having implicit ordering

File: src/sentry/apidocs/api_ownership_stats_dont_modify.json
Comment (schew2381) [inline src/sentry/apidocs/api_ownership_stats_dont_modify.json:1]: You can ignore this file change, it's here b/c the github action to generate this JSON file is not being run in other PRs, so the changes show up here

File: src/sentry/apidocs/hooks.py [hunk 2]
```diff
@@ -197,18 +198,44 @@ def custom_postprocessing_hook(result: Any, generator: Any, **kwargs: Any) -> An
...
+                            sorted(
+                                schema["properties"].items(),
+                                key=lambda param: 0 if param[0] in required else 1,
+                            )
+                        )
+                except KeyError as e:
+                    raise SentryApiBuildError(
+                        f"Unable to parse body parameters due to KeyError {e} for endpoint {endpoint_name}. Please reach out to Enterprise team to fix."
```
Comment (sentaur-athena) [inline src/sentry/apidocs/hooks.py:237]: #discuss-apis is a better place or ownersapi github group.

**Review by sentaur-athena — APPROVED** [review]
I agree to explicitly sorting. Just a nit 😅

**Pull request merged**